pkgs/top-level/python-packages.nix

```diff
@@ -2434,8 +2434,6 @@ self: super: with self; {
 
   cocotb-bus = callPackage ../development/python-modules/cocotb-bus { };
 
-  codecov = callPackage ../development/python-modules/codecov { };
```

Comment:
we should raise a trace output here which describes the reasoning. additionally it would be good to show up a migration path on how to patch package description to remove codecov as a dependency

Reply:
Suggestion:

```nix
  codecov = throw "codecov has been removed, as it was deprecated and broken since February 2022"; # added 2024-10-24
```